src/protocols/radius/base.c

```diff
@@ -517,7 +517,7 @@ bool fr_radius_ok(uint8_t const *packet, size_t *packet_len_p,
 	 *
 	 *	i.e. No response to the NAS.
 	 */
-	if (packet_len < totallen) {
+	if (totallen > packet_len) {
```

Comment:
Why does changing the order of comparisons help?  Is Coverity that stupd?

Reply:
Found the case; it's CID 1503921. The code in question (starting at src/protocols/dns/base.c:235)

`

				/*
				 *	Can't point to the header.
				 */
				if (offset < 12) {
					DECODE_FAIL(POINTER_TO_HEADER);
					return false;
				}

				/*
				 *	Can't point to the current label.
				 */
				if ((packet + offset) >= start) {
					DECODE_FAIL(POINTER_LOOPS);
					return false;
				}

				/* coverity[tainted_data] */
				if (!fr_dns_marker[offset]) {
					DECODE_FAIL(POINTER_TO_NON_LABEL);
					return false;
				}
`

That insists that offset be in [12, start - packet), but coverity doesn't realize it and gives the tainted {data, scalar} error.